### ISSUE TITLE
Value Checker support for Math#min and Math#max

### ISSUE THREAD

**Issue by kelloggm** [typetools/checker-framework#1687]
The Value Checker has no annotations on `java.util.Math#min` and `java.util.Math#max`, resulting in false positives in computations involving those functions. Consider:

```java
void mathTest(@IntRange(from = 0, to = 20) int x, @IntRange(from = 5, to = 15) int y) {
     @IntRange(from = 0, to = 5) int min = Math.min(x, y);
     @IntRange(from = 15, to = 20) int max = Math.max(x, y);
}
```
Expected: no warnings
Result:
```java
tests/value/MathMinMax.java:5: error: [assignment.type.incompatible] incompatible types in assignment.
        @IntRange(from = 0, to = 5) int min = Math.min(x, y);
                                                      ^
  found   : @UnknownVal int
  required: @IntRange(from=0, to=5) int
tests/value/MathMinMax.java:6: error: [assignment.type.incompatible] incompatible types in assignment.
        @IntRange(from = 15, to = 20) int max = Math.max(x, y);
                                                        ^
  found   : @UnknownVal int
  required: @IntRange(from=15, to=20) int
2 errors
```

**Comment by kelloggm** [comment]
This is a relatively minor problem, but it came up once in an Index Checker case study.

**Comment by wmdietl** [comment]
I don't understand the proposal.
When I call `mathTest(20, 15)` the local variable `min` will have value 15.
When I call `mathTest(0, 5)` the local variable `max` will have value 5.
So the given annotations on the two local variables seem wrong.

I would guess for `Math.min` it should take the minimum of the `from` bounds as resulting `from`; and the minimum of the `to` bounds as resulting `to`, not just the two `from` bounds. Similarly for max.

**Comment by kelloggm** [comment]
You're right; this is my mistake. The correct test case should be:

```java
void mathTest(@IntRange(from = 0, to = 20) int x, @IntRange(from = 5, to = 15) int y) {
     @IntRange(from = 0, to = 15) int min = Math.min(x, y);
     @IntRange(from = 5, to = 20) int max = Math.max(x, y);
}
```
